src/resolvers/mutations/CreateUser.ts

```diff
@@ -22,9 +29,48 @@ export const CreateUser = (_, { user: CreateUserInput }, { request, response }):
       throw Error('Malformed token payload');
     }
 
-    return { id: 1, name: 'name', email: 'email', birthDate: new Date(), cpf: '10020030012', password: ''};
+    let errorMessage = '';
+    const isEmailUnique = !(await db.manager.findOne(User, { email: user.email }));
+    if (!isEmailUnique) {
+      errorMessage += 'Email already used';
+    }
+    
+    const verificationResult: PasswordVerification[] = verifyPassword(user.password);
+    const isValidPassword: boolean = verificationResult
+      .map((passwordVerification: PasswordVerification) => passwordVerification.valid)
+      .reduce((previousValue: boolean, currentValue: boolean) => previousValue && currentValue);
+
+    if (!isValidPassword) {
+      verificationResult.forEach((passwordVerification: PasswordVerification) => {
+        if (!passwordVerification.valid) {
+          errorMessage += `\n${passwordVerification.errorMessage}`;
+        }
+      });
+    }
+
+    if (errorMessage.length > 0) {
+      throw Error(errorMessage);
```

Comment:
Quando você perguntou se era melhor enviar um erro só ou todos, eu achei que você tava falando só da validação dos campos do input da mutation. Na minha opinião, uma vez que você detectou que o e-mail já não existe, poderia jogar o erro direto.

Reply:
De qualquer forma, se você for seguir com o tratamento de mais de um erro simultâneo, creio que seria interessante você retornar uma lista de erros então, em vez de uma mensagem que é uma string quebrada em linhas. Consegue pensar por que seria melhor assim?

Reply:
Acredito que seria melhor uma lista de erros porque o client (App ou site) conseguiria iterar na lista e decidir como seria melhor apresentar para o usuário, sendo uma implementação mais flexível.